docs/utilities.rst

```diff
@@ -1,25 +1,185 @@
 Developer utilities
 ===================
 
+
+This section provides a detailed description of some commonly used utilities for Taichi developers.
+
 Logging
```

Comment:
Hi, sorry about dismissing your changes, I have already done with `Logging` in #1475 :)

Reply:
May you merge these 2 changes into a single one? I also have no idea about how to deal with this conflict.

Reply:
To prevent conflicts with #1475, you may move the `Profiler` to another section like `profiler.rst` :) I'll try to merge conflicts in `Logging` section once one of these two PR is merged.